config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -124,18 +133,18 @@ public void deleteConfig(
       DeleteConfigRequest request, StreamObserver<DeleteConfigResponse> responseObserver) {
     try {
       ConfigResourceContext configResourceContext = getConfigResourceContext(request);
-      ContextSpecificConfig configToDelete = configStore.getConfig(configResourceContext);
+      Optional<ContextSpecificConfig> maybeConfig = configStore.getConfig(configResourceContext);
```

Comment:
suggest mapping this one to a response object. Then, if response is present, delete configs and return; else error.

Reply:
i didnt get what you mean by this?

Reply:
e.g.
```java
    try {
      ConfigResourceContext configResourceContext = getConfigResourceContext(request);
      DeleteConfigResponse deleteResponse = configStore.getConfig(configResourceContext)
          .map(configToDelete -> DeleteConfigResponse.newBuilder().setDeletedConfig(configToDelete).build())
          .orElseThrow(Status.NOT_FOUND::asException);

      // delete the config for the specified config resource.
      configStore.deleteConfigs(Set.of(configResourceContext));
      responseObserver.onNext(deleteResponse);
      responseObserver.onCompleted();
    } catch (Exception e) {
      log.error("Delete config failed for request:{}", request, e);
      responseObserver.onError(e);
    }
```

Reply:
got it